### PR TITLE
Restore react component method signature

### PR DESCRIPTION
**Pull request by vegetabill**
change it back to the official gem's signature
accommodate our extra js as an option

### CODE REVIEW

File: lib/react/rails/view_helper.rb [hunk 1]
```diff
@@ -14,43 +14,44 @@ def react_javascript(&block)
...
+        html_options.except!(:tag, :prerender, :store_init_js)
 
         initialization_js_tags + content_tag(html_tag, '', html_options, &block)
       end
 
       private
 
-      #
-      def init_js_tags(component_name:, props: {}, options: {}, store_update_calls: "")
-        if store_update_calls.present? && !store_update_calls.html_safe?
```
Comment (dan-brewster) [inline lib/react/rails/view_helper.rb:51]: I'm a little concerned about dropping the html_safe? check here. I'm not sure there is a drawback, but if it helps people avoid putting unsafe'd JS here, I think that's a plus.

File: lib/react/rails/view_helper.rb [hunk 2]
```diff
@@ -14,43 +14,44 @@ def react_javascript(&block)
...
+        if store_init_js.present? || (request && request.xhr?)
           initialization_js_tags = init_js_tags(
-                                      component_name: component_name,
+                                      name: name,
                                       props: props,
                                       options: options,
-                                      store_update_calls: store_update_calls)
+                                      store_init_js: store_init_js)
         end
-        options = {:tag => options} if options.is_a?(Symbol)
```
Comment (dan-brewster) [inline lib/react/rails/view_helper.rb:31]: Previously, you could replace the options hash with just ":span" or whatever, and it plugged that in for the tag. I assumed that behavior was inherited from the master branch.

I'm not saying we need to keep it, but just want to check if this was a conscious removal of that functionality.

Comment (vegetabill) [inline lib/react/rails/view_helper.rb:31]: It was just moved to line 24.

I think this also means ReviewBoard is better than GitHub ;-)

File: lib/react/rails/view_helper.rb (hunk 1 above)
Comment (vegetabill) [inline lib/react/rails/view_helper.rb:51]: I don't think it makes sense to check in the private method when we're already checking in the public one.

We could _only_ check in the private one if you think that makes more sense?

File: lib/react/rails/view_helper.rb (hunk 2 above)
Comment (dan-brewster) [inline lib/react/rails/view_helper.rb:31]: Ah -- I see it now. Looks good.

File: lib/react/rails/view_helper.rb (hunk 1 above)
Comment (dan-brewster) [inline lib/react/rails/view_helper.rb:51]: Nope, that's the right thing. I didn't see that this moved, either.

**Comment by wwcline** [comment]
Looks good to me.

**Pull request merged**